src/ServiceBus.AttachmentPlugin/AzureStorageAttachment.cs

```diff
@@ -70,10 +77,26 @@ static void SetValidUntil(ICloudBlob blob, TimeSpan timeToBeReceived)
         public override async Task<Message> AfterMessageReceive(Message message)
         {
             var userProperties = message.UserProperties;
+
             if (!userProperties.ContainsKey(configuration.MessagePropertyToIdentifyAttachmentBlob))
             {
                 return message;
             }
+            
+            //If the message contains a property with a SAS uri we download using that information
+            if (userProperties.ContainsKey(configuration.MessagePropertyForSasUri))
+            {
+                var blobFromSasUri = new CloudBlockBlob(new Uri(userProperties[configuration.MessagePropertyForSasUri].ToString()));
+                using (var ms = new MemoryStream())
```

Comment:
You don't really need a memory stream here as we need a `byte[]`. Can do the following:

```c#
await blob.FetchAttributesAsync().ConfigureAwait(false);
var bytes = new byte[blob.Properties.Length];
await blob.DownloadToByteArrayAsync(bytes, 0).ConfigureAwait(false);
message.Body = bytes;
```